graph/vertex_cover.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+  @author [Rajat Garg](https://github.com/Rajat379)
+  @file
+
+  Implementation of a vertex cover in a graph.
+*/
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <list>
+
+namespace {
+class Graph {
+    int V;
+    std::list<int> *adj;  // Pointer to an array containing adjacency lists
+ public:
+    Graph(int V);
+    void addEdge(int v, int w);  // function to add an edge to graph
+    void printVertexCover();     // prints vertex cover
+};
+
+}  // namespace
+
+Graph::Graph(int V) {
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.